source/opt/set_spec_constant_default_value_pass.cpp

```diff
@@ -98,11 +98,15 @@ std::vector<uint32_t> ParseDefaultValueBitPattern(
     }
     return result;
   } else if (const auto* IT = type->AsInteger()) {
-    if (IT->width() == input_bit_pattern.size() * sizeof(uint32_t) * 8) {
+    auto width = IT->width();
+    if (width == 8 || width == 16) width = 32;
+    if (width == input_bit_pattern.size() * sizeof(uint32_t) * 8) {
```

Comment:
This is good.

I'd also go with a generalization:

  width = min(width, 32); 

One thing to watch for is that SPIR-V requires literals to fill out upper (unused) bits in a certain way.  From the definition of Literal:

> For a numeric literal, the lower-order words appear first. If a numeric type’s bit width is less than 32-bits, the value appears in the low-order bits of the word, and the high-order bits must be 0 for a floating-point type or integer type with Signedness of 0, or sign extended for an integer type with a Signedness of 1 (similarly for the remaining bits of widths larger than 32 bits but not a multiple of 32 bits).
 
So it's possible to send in invalid data through this API now.
Given that this is a net improvement, I'll approve.